### PR TITLE
Add backup and restore support

### PR DESCRIPTION
**Pull request by ScottVerbeek**

### CODE REVIEW

File: backup/moodle2/restore_local_extension_plugin.class.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+ * @package    local_extension
+ * @copyright  2024 Scott Verbeek <scottverbeek@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class restore_local_extension_plugin extends restore_local_plugin {
+
+    public function define_module_plugin_structure() {
+        $paths = [];
+
+        if(!$this->task->get_setting('userinfo')) {
```
Comment (rhell4) [inline backup/moodle2/restore_local_extension_plugin.class.php]: Missing space after if

File: backup/moodle2/restore_local_extension_plugin.class.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        // Add file to restore.
+        restore_dbops::send_files_to_pool(
+            $this->task->get_basepath(),
+            $this->task->get_restoreid(),
+            $component,
+            $filearea,
+            $data->oldcontextid,
+            $data->userid,
+            $itemname,
+            $data->olditemid, $newcontext->id,
```
Comment (rhell4) [inline backup/moodle2/restore_local_extension_plugin.class.php]: $newcontext->id should be on a new line.

**Review by rhell4 — CHANGES_REQUESTED** [review]
A few other things need cleaning up;
- [x] Doc blocks would be nice
- [x] Would be nice for the global $DB inclusion in `backup_local_extension_plugin.class.php` to be at the top of the function, just makes it more visible and consistent
- [x] There's a bunch of double or triple spaces in `restore_local_extension_plugin.class.php`, these should either be single spaces or make the = lined up

**Comment by ScottVerbeek** [comment]
Hi @rhell4  thanks for the review. I have amended the items listed.

**Review by rhell4 — APPROVED** [review]

**Pull request merged**